### PR TITLE
Add an example versioned module using google and google-beta

### PR DESCRIPTION
**Pull request by rileykarson**
Writing modules against tpg/tpgb is hard - introduce an example of how to use beta features with both providers in a single module.

### CODE REVIEW

**Comment by chrisst** [comment]
⚠ you're still against 2.0.0

**Comment by rileykarson** [comment]
Thanks! I'd pushed the change so the commit was properly based but had forgotten to click the UI button.

File: examples/example-versioned-module/ip/main.tf [hunk 1]
```diff
@@ -0,0 +1,16 @@
+resource "google_compute_address" "ip_address" {
+  # We'll only generate this block if the value of
+  # has_labels is 0! Effectively an if statement.
+  count = "${1 - local.has_labels}"
+
+  name = "${var.name}"
+}
+
+resource "google_compute_address" "ip_address_beta" {
```
Comment (morgante) [inline examples/example-versioned-module/ip/main.tf]: @rileykarson I'm confused how this works- shouldn't this resource need to specify the beta provider?

Comment (rileykarson) [reply examples/example-versioned-module/ip/main.tf]: See [this example of invoking it](https://github.com/terraform-providers/terraform-provider-google/pull/2709/files#diff-93659ab1ad57330040aa5f8dcc7254f3R10); while normally aliased provider names are namespaced eg `provider = "google.alias"`, we're able to override the `google` provider in the module [with the providers block](https://github.com/terraform-providers/terraform-provider-google/pull/2709/files#diff-93659ab1ad57330040aa5f8dcc7254f3R20).

Using `provider = "google-beta"` explicitly throughout the module is probably what I would advise now, although a mechanism like this works to share a single config w/ both provider versions at the cost of needing to use a `providers` block to use beta features.

Comment (morgante) [reply examples/example-versioned-module/ip/main.tf]: Yeah, `provider = "google-beta"` is what we've settled on as well. Maybe we should update this example?

Comment (rileykarson) [reply examples/example-versioned-module/ip/main.tf]: I think it's worth keeping this as a distinct example of a module that can support either version; see the README: https://github.com/terraform-providers/terraform-provider-google/pull/2709/files#diff-b56d3749183c558a9af3bf22e0ef72b7R10

**Pull request merged**